File: java/eyes.selenium.common/src/main/java/com/applitools/eyes/selenium/fluent/AccessibilityRegionByElement.java
package com.applitools.eyes.selenium.fluent;

import com.applitools.eyes.AccessibilityRegionType;
import com.applitools.eyes.fluent.GetRegion;
import com.applitools.eyes.fluent.IGetAccessibilityRegionType;
import com.applitools.eyes.serializers.WebElementSerializer;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.openqa.selenium.WebElement;

public class AccessibilityRegionByElement implements GetRegion, IGetAccessibilityRegionType {

    protected final AccessibilityRegionType regionType;
    @JsonSerialize(using = WebElementSerializer.class)
    protected final WebElement element;

    public AccessibilityRegionByElement(WebElement element, AccessibilityRegionType regionType) {
        this.element = element;
        this.regionType = regionType;
    }

    @Override
    public AccessibilityRegionType getAccessibilityRegionType() {
        return regionType;
    }


    @JsonProperty("element")
    public WebElement getElement() {
        return element;
    }

}

File: dotnet/Eyes.Appium.DotNet/Fluent/AccessibilityRegionByElement.cs
using System.Collections.Generic;
using Applitools.Appium.Utils;
using Applitools.Fluent;
using OpenQA.Selenium;

namespace Applitools.Appium
{
    internal class AccessibilityRegionByElement : IGetAccessibilityRegion, IGetAppiumRegion, IGetAccessibilityRegionType
    {
        private readonly AccessibilityRegionType regionType_;
        private IWebElement element_;

        public AccessibilityRegionByElement(IWebElement element, AccessibilityRegionType regionType)
        {
            element_ = element;
            regionType_ = regionType;
        }

        AccessibilityRegionType IGetAccessibilityRegionType.AccessibilityRegionType => regionType_;
        
        public TAccessibilityRegion ToRegion()
        {
            return new ElementAccessibilityRegion
            {
                Type = regionType_,
                Region = new RegionElement
                {
                    //ElementId = element_.GetElementId() // ?
                }
            };
        }

        IList<IWebElement> IGetAppiumRegion.GetElements(IWebDriver driver)
        {
            return new List<IWebElement> { element_ };
        }
    }
}